src/bin/src/systems/chunk_sender.rs

```diff
@@ -0,0 +1,132 @@
+use crate::systems::definition::System;
+use async_trait::async_trait;
+use ferrumc_core::identity::player_identity::PlayerIdentity;
+use ferrumc_core::transform::position::Position;
+use ferrumc_net::connection::StreamWriter;
+use ferrumc_net::packets::outgoing::chunk_and_light_data::ChunkAndLightData;
+use ferrumc_net::packets::outgoing::set_center_chunk::SetCenterChunk;
+use ferrumc_net::GlobalState;
+use ferrumc_net_codec::encode::NetEncodeOpts;
+use std::ops::Div;
+use std::simd::num::SimdFloat;
+use std::simd::{f64x2, StdFloat};
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::time::Duration;
+use tracing::{debug, error, info};
+
+const CHUNK_RADIUS: i32 = 12;
+
+pub(super) struct ChunkSenderSystem {
+    pub stop: AtomicBool,
+}
+
+impl ChunkSenderSystem {
+    pub const fn new() -> Self {
+        Self {
+            stop: AtomicBool::new(false),
+        }
+    }
+}
+
+#[async_trait]
+impl System for ChunkSenderSystem {
+    async fn start(self: Arc<Self>, state: GlobalState) {
+        info!("Chunk sender system started");
+
+        while !self.stop.load(Ordering::Relaxed) {
+            let players = state
+                .universe
+                .query::<(&PlayerIdentity, &Position, &mut StreamWriter)>();
+            // TODO: This is so ass. Please fix this.
+            for (_entity, (player, position, mut conn)) in players {
```

Comment:
And like I mentioned, don't get a `mut conn` lock here. It will block every other operation that requires the connection. Instead use the `entity` (id) and get the `<StreamWriter>` when you're ready to send the chunks.

Reply:
Another thing is, this will eventually be not very useful. We will need to implement a system that dynamically sends chunks based on `Δ pos`. Like we did in v2. If the last chunk send position vs now is a threshold apart, then send the chunks.

Reply:
Yeah I'm aware and have said several times. This is a shit system and will need to be redone, but it does an ok enough job for now, but I just wanna finish up the rewrite and fixing this will push it back a while.